src/core/vdom/helpers/normalize-children.js

```diff
@@ -45,7 +45,9 @@ function normalizeArrayChildren (children: any, nestedIndex?: string): Array<VNo
     last = res[res.length - 1]
     //  nested
     if (Array.isArray(c)) {
-      res.push.apply(res, normalizeArrayChildren(c, `${nestedIndex || ''}_${i}`))
+      // avoid elements to be set key, which created by slot
+      const subNestedIndex = isTrue(c._rendered) ? undefined : `${nestedIndex || ''}_${i}`
```

Comment:
@pengchongfu 

I did not find any problems in the ` render function`

html:
```html
 <div id="app"></div>
```
javascript: 
``` javascript
var vm = new Vue({
  el: '#app',
  data: {
    count: 1
  },
  render: function (h) {
    var list = []

    for (var i = 0; i < this.count; i++) {
      list.push(h('span', [this.count]))
    }

    return h('div', [
      list,
      h('input', {attrs: { 
        value: 'a',
        type: 'text'
      }})
    ])
  }
})
```
You can input something in the input box,
then modify the value of `count` in the console

and input`s value did not changed